src/gax.ts

```diff
@@ -67,27 +67,53 @@ import {toLowerCamelCase} from './util';
 
 /**
  * Per-call configurable settings for retrying upon transient failure.
+ * @implements {RetryOptionsType}
  * @typedef {Object} RetryOptions
- * @property {String[]} retryCodes
+ * @property {String[] | (function)} retryCodesOrShouldRetryFn
  * @property {BackoffSettings} backoffSettings
+ * @property {(function)} getResumptionRequestFn
  */
 export class RetryOptions {
-  retryCodes: number[];
+  retryCodesOrShouldRetryFn: number[] | ((error: any) => boolean);
   backoffSettings: BackoffSettings;
-  constructor(retryCodes: number[], backoffSettings: BackoffSettings) {
-    this.retryCodes = retryCodes;
+  getResumptionRequestFn?: (response: any) => any;
```

Comment:
Can we use better types here? There are interfaces like `RequestType` and `ResponseType` defined somewhere (in client interface maybe?) that are less generic than `any` and have some semantic meening.

Reply:
I am totally in favor but am running into TS issues - sent you a chat about what's going on

Reply:
Just to document here for other folks who may look at this PR, there were some issues with using `RequestType` and `ResponseType` with the test-application, so we changed them to `{}` as a compromise

Reply:
Done!